sql/planbuilder/from.go

```diff
@@ -129,54 +138,101 @@ func (b *Builder) buildJoin(inScope *scope, te *ast.JoinTableExpr) (outScope *sc
 	return outScope
 }
 
-// buildNaturalJoin logically converts a NATURAL_JOIN to an INNER_JOIN.
-// All column names shared by the two tables are used as equality filters
-// in the join. The intersection of all unique columns is projected.
-// Common table attributes are rewritten to reference the left definitions
-// //.
-// NATURAL_JOIN(t1, t2)
-// =>
-// PROJ(t1.a1, ...,t1.aN) -> INNER_JOIN(t1, t2, [t1.a1=t2.a1,..., t1.aN=t2.aN])
-func (b *Builder) buildNaturalJoin(inScope, leftScope, rightScope *scope) (outScope *scope) {
+// buildUsingJoin converts a JOIN with a USING clause into an INNER JOIN, LEFT JOIN, or RIGHT JOIN; NATURAL JOINs are a
+// subset of USING joins.
+// The scope of these join must contain all the qualified columns from both left and right tables. The columns listed
+// in the USING clause must be in both left and right tables, and will be redirected to
+// either the left or right table.
+// An equality filter is created with columns in the USING list. Columns in the USING
+// list are de-duplicated and listed first (in the order they appear in the left table), followed by the remaining
+// columns from the left table, followed by the remaining columns from the right table.
```

Comment:
i still like the SQL query rewrite examples, personally. SQL before and after is easy to read